### PR TITLE
Add a module and class for handling sky grids for `pycbc_multi_inspiral`

### PR DESCRIPTION
**Pull request by titodalcanton**
Calculating the coherent network SNR (`pycbc_multi_inspiral`) requires a grid of points in the sky. This adds a module and class to manage such grids in a more centralized way.

## Standard information about the request

This is mostly refactoring existing code into a new module. This change affects PyGRB but does not change any output.

## Motivation

`pycbc_multi_inspiral` is a very long script at this point, and `pycbc_make_sky_grid` might also become very long as we implement various algorithms for placing sky grids available in the literature. Centralizing the management of sky grids, including reading/writing them from/to files, seems beneficial.

## Contents

I added a pycbc.tmpltbank.sky_grid module which currently contains a `SkyGrid` class, with methods to read and write sky grids from HDF5 files, calculate the antenna patterns and time delays over the grid for a network of detectors, and handle CLI arguments related to sky grids.

## Links to any issues or associated PRs

None.

## Testing performed

I wrote a short script to plot the antenna patterns and time delays over a sky grids of random points for various detectors, and visually checked that those made sense:

image

## Additional notes

None.

- [x] The author of this pull request confirms they will adhere to the [code of conduct](https://github.com/gwastro/pycbc/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by titodalcanton** [comment]
Note that I posed a couple design questions in the constructor of `SkyGrid`, and that I want to add explicit units and angular convention to the docstring, which I have not done yet.

File: pycbc/tmpltbank/sky_grid.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+        Parameters
+        ----------
+        ra: iterable of floats
+            Right ascensions for each point UNITS AND ANGULAR CONVENTION.
+        dec: iterable of floats
+            Declination for each point UNITS AND ANGULAR CONVENTION.
+        """
+        # Question: should the list of detectors and reference times be part of
+        # the SkyGrid object? Or should they be given each time they are needed,
```
Comment (spxiwh) [inline pycbc/tmpltbank/sky_grid.py:25]: I think I would store these in the SkyGrid object. I guess the issue if there are other, not PyGRB, use cases for this, but the class could always be refactored to accommodate such needs if this becomes relevant.

File: pycbc/tmpltbank/sky_grid.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        # independently? This decision will affect the `calculate_*()` methods
+        # down below.
+        # We store the points in a 2D array internally, first dimension runs
+        # over the list of points, second dimension is RA/dec.
+        # Question: should we use Astropy sky positions instead?
```
Comment (spxiwh) [inline pycbc/tmpltbank/sky_grid.py:30]: No preference on this from my perspective

**Review by spxiwh — COMMENTED** [review]
Francesco should probably give final approval for this, but it all looks good to me. I left comments in response to the questions raised.

File: pycbc/tmpltbank/sky_grid.py (hunk 1 above)
Comment (titodalcanton) [reply pycbc/tmpltbank/sky_grid.py:25]: Sounds good.

**Comment by titodalcanton** [comment]
@pannarale this is now ready for your review.

**Review by pannarale — APPROVED** [review]

**Pull request merged**